GraphColoring/Tests.qs

```diff
@@ -267,4 +272,103 @@ namespace Quantum.Kata.GraphColoring {
             Message($"Got correct coloring {coloring}");
         }
     }
+
+    @Test("QuantumSimulator")
+    operation T31_DoesEdgeStartWithVertex () : Unit {
+        Fact(DoesEdgeStartWithVertex( (1,2), 1) == true,
+             $"Edge (1,2) judged incorrect for vertex 1");
+        Fact(DoesEdgeStartWithVertex( (1,2), 2) == false,
+             $"Edge (1,2) judged correct for vertex 2");
+        Fact(DoesEdgeStartWithVertex( (1,2), 3) == false,
+             $"Edge (1,2) judged correct for vertex 2");
+    }
+
+    @Test("QuantumSimulator")
+    operation T321_IsVertexWeaklyColored () : Unit {
+        let testCases = ExampleGraphs();
+        let colorings = [[0, 0, 0],
+                         [3, 2, 0, 0],
+                         [1, 0, 1, 2, 1],
+                         [0, 1, 1, 1, 1],
+                         [0, 1, 2, 2, 2]
+                        ];
+        let expectedResults = [[true, true, true],
+                               [true, true, false, true],
+                               [true, true, true, true, false],
+                               [true, false, true, false, false],
+                               [true, true, false, false, true]
+                              ];
+        for t in 0..Length(testCases)-2 {
+            let (V_t, edges_t) = testCases[t];
+            for vertex in 0 .. V_t - 1 {
+            Fact(IsVertexWeaklyColored(V_t, edges_t, colorings[t], vertex) == expectedResults[t][vertex],
+                $"Vertex = {vertex} judged {not expectedResults[t][vertex]} for coloring = {colorings[t]} , edges = {edges_t}");
+            }
+        }
+    }
+
+    @Test("QuantumSimulator")
+    operation T322_IsWeakColoringValid () : Unit {
+        let testCases = ExampleGraphs();
+        let exampleColoringForThreeVertices = [[0, 0, 0], [2, 1, 3]];
+        let exampleColoringForFourVertices = [[0, 2, 1, 3], [3, 2, 0, 0], [0, 2, 1, 0] ];
+        let exampleColoringForFiveVertices = [[0, 1, 2, 3, 1], [0, 1, 1, 2, 2], [0, 0, 1, 2, 3]];
+
+
+        let (V0, edges0) = testCases[0];
+        for (coloring, expectedResult) in Zipped(exampleColoringForThreeVertices, [true, true]) {
```

Comment:
Nice use of Zipped to avoid repeating facts for each graph! I think you can make it even nicer if you create an array of colorings and expected verdicts for each graph and zip that array with ExampleGraphs(), then just iterate over that big array checking each entry.

So something along the lines of 
```qsharp
let coloringsAndVerdicts3 = Zipped(exampleColoringForThreeVertices, [true, true]);
...
let fullTestCases = Zipped(testCases, [coloringsAndVerdicts3, coloringsAndVerdicts4, ...]);
```

Reply:
Done. Thanks for this suggestion. Made the code lot more cleaner